test/e2e/cypress/integration/profiles/oidc-provider/login.spec.ts

```diff
@@ -146,3 +147,189 @@ context("OpenID Provider", () => {
     cy.getCookie("ory_hydra_session_dev").should("be.null")
   })
 })
+
+context("OpenID Provider - change between flows", () => {
```

Comment:
E2E test which verifies if the `return_to` URL is set when switching between flows after initializing an OAuth2 flow. 

The AX takes the current `return_to` query parameter from the login request and passes it along to the recovery flow. So there is some UI logic that makes this possible. The only thing kratos does is take the current request URL and checks if it has a `login_challenge` query parameter. Based on this parameter it overwrites the `return_to` parameter.